core/src/ops/array/concat.rs

```diff
@@ -4,15 +4,39 @@ use crate::ops::prelude::*;
 
 #[derive(Debug, Clone, new)]
 pub struct Concat {
-    axis: usize,
+    axis: i64,
 }
 
 impl Concat {
+    fn resolve_axis(&self, rank: i64) -> TractResult<usize> {
+        let axis = {
+            let axis_res: TractResult<i64> = {
+                if 0 <= self.axis && self.axis <= rank - 1 {
+                    Ok(self.axis)
+                } else if -rank <= self.axis && self.axis < 0 {
+                    Ok(self.axis + rank)
+                } else {
+                    bail!("Illegal combination of values for rank and axis")
```

Comment:
Can we give the values in the error message (I would actually have been happy for a simpler check impl, but...)

Reply:
See the one in reduce...